apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -437,13 +455,28 @@ private Mono<Long> mapAndInsert(
 
             int i = 0;
             for (DatasetItem item : items) {
+                Map<String, String> inputData = getOrDefault(item.inputData());
+
+                String input = getOrDefault(item.input());
+                String expectedOutput = getOrDefault(item.expectedOutput());
+
+                if (!inputData.containsKey("input") && StringUtils.isNotBlank(input)) {
+                    inputData.put("input", JsonUtils.writeValueAsString(new JsonValue(item.input())));
+                }
+
+                if (!inputData.containsKey("expected_output") && StringUtils.isNotBlank(expectedOutput)) {
+                    inputData.put("expected_output",
+                            JsonUtils.writeValueAsString(new JsonValue(item.expectedOutput())));
+                }
```

Comment:
I'm not convinced about the value of the old deprecated fields overwriting the value here. We need to think about the rollout strategy a bit here, considering that multiple versions of the SDK are going to be co-existing and that the FE might just do the resolution logic until the old fields are deprecated.

Reply:
This would avoid leaving the migration up to the user; it's more of a convenience to avoid breaking existing code.

Reply:
Agreed to do this, in order to encapsulate the resolution logic in the BE.